reposerver/repository/repository.go

```diff
@@ -98,6 +98,37 @@ func NewService(metricsServer *metrics.MetricsServer, cache *reposervercache.Cac
 	}
 }
 
+// List a subset of the refs (currently, branches and tags) of a git repo
+func (s *Service) ListRefs(ctx context.Context, q *apiclient.ListRefsRequest) (*apiclient.Refs, error) {
+	gitClient, err := s.newClient(q.Repo)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO check cache
+
+	s.metricsServer.IncPendingRepoRequest(q.Repo.Repo)
+	defer s.metricsServer.DecPendingRepoRequest(q.Repo.Repo)
+
+	// TODO is locking necessary here? We are not modifying the repository, just running an ls-remote
```

Comment:
Locking is not required ( it is necessary only if we are touching local repo files). Please remove comment